### ISSUE TITLE
Spring Turbine Stream does not get messages from hystrixStreamOutput

### ISSUE THREAD

**Issue by fdimauro** [spring-cloud/spring-cloud-netflix#3504]
Spring Hystrix Steam (org.springframework.cloud.netflix.hystrix.stream) is currently writing it's hystrix messages to the hystrixStreamOutput exchange.

The current revision of Netflix Turbine Stream (org.springframework.cloud.netflix.turbine.stream) has a ServiceActivator listening on the 
turbineStreamInput queue.

So the aggregator never gets the messages from the rabbit exchange. 

I tried adding:

spring:
  cloud:
    stream:
      bindings:
        turbineStreamInput:
          destination: hystrixStreamOutput

But still I see no turbineStreamInput queue created, nor do I see ANY queue bound to the hystrixStreamOutput exchange.

So my question is how do I get the Netflix Turbine Stream ServiceActivator to get the messages that are being placed on the hystrixStreamOutput rabbit exchange?

**Comment by spencergibb** [comment]
What versions are you using? What are your dependencies?

**Comment by fdimauro** [comment]
<spring.cloud.version>Greenwich.RELEASE</spring.cloud.version>

**Comment by spencergibb** [comment]
Can you retry with the latest Greenwich.SR1 and the latest boot?

**Comment by fdimauro** [comment]
Yes sir, will do!

**Comment by fdimauro** [comment]
Ok updated to 

<spring.cloud.version>Greenwich.SR1</spring.cloud.version>
<spring.boot.version>2.1.4.RELEASE</spring.boot.version>

I see the service activator subscribe to the channel:

 2019-04-16 10:51:57,278  INFO --- [main] o.s.i.endpoint.EventDrivenConsumer:107  : Adding {service-activator:hystrixStreamAggregator.sendToSubject.serviceActivator} as a subscriber to the 'turbineStreamInput' channel  
 2019-04-16 10:51:57,279  INFO --- [main] o.s.c.s.m.DirectWithAttributesChannel:64  : Channel 'hystrix-aggregator-1.turbineStreamInput' has 1 subscriber(s).  
 2019-04-16 10:51:57,279  INFO --- [main] o.s.i.endpoint.EventDrivenConsumer:159  : started hystrixStreamAggregator.sendToSubject.serviceActivator  

I also can see hystrix messages that are being sent to the hystrixStreamOutput exchange:

{"origin":{"host":"10.0.0.191","port":9001,"serviceId":"powerdata-api-svc"},"event":"message","data":{"type":"HystrixCommand","name":"powerdata-api-svc.userInfo","group":"UserProxyService","currentTime":1555434150974,"isCircuitBreakerOpen":false,"errorPercentage":0,"errorCount":0,"requestCount":0,"rollingCountCollapsedRequests":0,"rollingCountExceptionsThrown":0,"rollingCountFailure":0,"rollingCountFallbackFailure":0,"rollingCountFallbackRejection":0,"rollingCountFallbackSuccess":0,"rollingCountResponsesFromCache":0,"rollingCountSemaphoreRejected":0,"rollingCountShortCircuited":0,"rollingCountSuccess":0,"rollingCountThreadPoolRejected":0,"rollingCountTimeout":0,"currentConcurrentExecutionCount":0,"latencyExecute_mean":35,"latencyExecute":{"0":27,"25":27,"50":33,"75":41,"90":42,"95":42,"99":42,"99.5":42,"100":42},"latencyTotal_mean":35,"latencyTotal":{"0":27,"25":27,"50":33,"75":41,"90":42,"95":42,"99":42,"99.5":42,"100":42},"propertyValue_circuitBreakerRequestVolumeThreshold":20,"propertyValue_circuitBreakerSleepWindowInMilliseconds":5000,"propertyValue_circuitBreakerErrorThresholdPercentage":50,"propertyValue_circuitBreakerForceOpen":false,"propertyValue_circuitBreakerForceClosed":false,"propertyValue_circuitBreakerEnabled":true,"propertyValue_executionIsolationStrategy":"SEMAPHORE","propertyValue_executionIsolationThreadTimeoutInMilliseconds":60000,"propertyValue_executionIsolationThreadInterruptOnTimeout":true,"propertyValue_executionIsolationThreadPoolKeyOverride":null,"propertyValue_executionIsolationSemaphoreMaxConcurrentRequests":200,"propertyValue_fallbackIsolationSemaphoreMaxConcurrentRequests":10,"propertyValue_metricsRollingStatisticalWindowInMilliseconds":10000,"propertyValue_requestCacheEnabled":true,"propertyValue_requestLogEnabled":true,"reportingHosts":1}}
--

But the turbineStreamInput never gets any messages and the sendToSubject method of HystrixStreamAggregator is never triggered.

**Comment by fdimauro** [comment]
Spencer,

Thanks for you time and help. I found the issue.

If the actuator library is included somewhere in there a class is derived from Binder.class. RabbitServiceAutoConfiguration has the following conditional:

@ConditionalOnMissingBean({Binder.class})

So the RabbitServiceAutoConfiguration class is never instantiated and no connections are made to RabbitMQ.

Removing the dependency on "spring-boot-starter-actuator" solved the problem!

**Comment by GissilyWork** [comment]
I have same issues when I updated to 
```xml
<spring.cloud.version>Greenwich.RELEASE</spring.cloud.version>
<spring.boot.version>2.1.0.RELEASE</spring.boot.version>
```
after I read above comment, I have tried to update to 
```xml
<spring.cloud.version>Greenwich.SR1</spring.cloud.version>
<spring.boot.version>2.1.4.RELEASE</spring.boot.version>
```
and removed the  dependency "spring-boot-starter-actuator" if existed, turbine stream is still not work.

According to my observation, when i am using 
```xml
<spring.cloud.version>Finchley.SR2</spring.cloud.version>
<spring.boot.version>2.0.9.RELEASE</spring.boot.version>
```
it is working, hystrix stream and turbine stream are using 
the same exchange "springCloudHystrixStream", turbine stream had got the events.
but after I updated to 2.1.4 and  Greenwich.SR1, 
I found if I have dependency "spring-boot-starter-actuator", 
hystrix stream sending to exchange "hystrixStreamOutput" and turbine stream subscribing from exhange "turbineStreamInput", then turbine stream got nothing.
If I removed dependency "spring-boot-starter-actuator", 
hystrix stream sending to exchange "hystrixStreamOutput",  and turbine stream subscribing from exhange "springCloudHystrixStream", then turbine stream still got nothing.
